### PR TITLE
Update Cypress to v12

### PR DESCRIPTION
**Pull request by d-rowe**
### Description
This is to update Cypress from v9 to v12. This PR is for the `cypress-unification` branch and these changes won't be merged into `main` until the final functional tests pass.

## Testing the changes
I've run the tests locally. We can use the results of the runner on this PR for verification as well.

## Changelog
- chore: update cypress to v12

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by zhongnansu — COMMENTED** [review]
CI is failing, could you fix? @d-rowe

**Comment by d-rowe** [comment]
> CI is failing, could you fix? @d-rowe

Yes, taking a look. The main culprit is

```
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/cypress-unification*:refs/remotes/origin/cypress-unification* +refs/tags/cypress-unification*:refs/tags/cypress-unification*
  The process '/usr/bin/git' failed with exit code 1
```

I'll work to repro locally and fix

File: cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
```
Comment (kavilla) [inline cypress.config.js:3]: older license

File: cypress.config.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
```
Comment (kavilla) [inline cypress.config.js:10]:
```suggestion
/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
```

File: cypress.config.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  viewportHeight: 1320,
+  env: {
+    openSearchUrl: 'http://localhost:9200',
+    SECURITY_ENABLED: false,
+    AGGREGATION_VIEW: false,
+    username: 'admin',
+    password: 'myStrongPassword123!',
+    ENDPOINT_WITH_PROXY: false,
+    MANAGED_SERVICE_ENDPOINT: false,
+    VISBUILDER_ENABLED: true,
```
Comment (kavilla) [inline cypress.config.js:28]: not related to this pr but if we have the chance we should structure these values a little bit more nicer

example: 
```
VISBUILDER: {
  ENABLED: true
}
```

File: cypress/support/e2e.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
```
Comment (kavilla) [inline cypress/support/e2e.js:9]: /*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

File: package.json [hunk 5]
```diff
@@ -384,7 +384,7 @@
     "chromedriver": "^121.0.1",
     "classnames": "^2.3.1",
     "compare-versions": "3.5.1",
-    "cypress": "9.5.4",
+    "cypress": "12.17.4",
```
Comment (kavilla) [inline package.json:387]: plugins if they dont run `yarn osd bootstrap --single-version=loose` might run into issues if they define the cypress version that is isn't this one

Comment (kavilla) [reply package.json:387]: something to consider mimicking 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2.x/package.json#L170 `opensearch-next`

Comment (kavilla) [reply package.json:387]: but i wouldn't block this pr since its going to feature branch but might need it for main

Comment (kavilla) [reply package.json:387]: build repo defines cypress version 12.13 https://github.com/opensearch-project/opensearch-build/blob/0931ccc2752d9eced6cb3015b822b4a10742fc80/docker/ci/dockerfiles/current/test.almalinux8.systemd-base.x64.arm64.ppc64le.dockerfile#L58

we might need to create an issue to bump the version in the build repo

Comment (kavilla) [reply package.json:387]: do we need to update this ? https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package.json#L59

File: cypress.config.js (hunk 2 above)
Comment (d-rowe) [reply cypress.config.js:4]: Thanks for providing the new one. I've updated this.

File: cypress.config.js (hunk 3 above)
Comment (d-rowe) [reply cypress.config.js:22]: Makes sense. I'll take a look and see if I can include that in this project in one of the subsequent PRs.

File: package.json (hunk 5 above)
Comment (d-rowe) [reply package.json:387]: Makes sense. Let me take a look at all the instances of cypress in our runners and other repos and I'll put together a small overview of what's needed to get all of these on a unified version of cypress.

**Review by kavilla — APPROVED** [review]

**Review by abbyhu2000 — APPROVED** [review]

**Pull request merged**